### ISSUE TITLE
Error - Allsky not installed

### ISSUE THREAD

**Issue by Jonk2** [AllskyTeam/allsky-modules#1]
Hi Alex, I've installed a fresh Dev branch of allsky and have come here to fetch the extra modules.

The clone has worked, chmod'd the install script and ran it.

I get the error message:

`AllSky does not seem to be installed. The /opt/allsky directory does not exist. Please install AllSky before installing the modules`

Allsky is installed, so what do I need to check?

![image](https://user-images.githubusercontent.com/77614536/219901707-5c657174-9f5f-4e91-979b-422d7c8512bd.png)

**Comment by Jonk2** [comment]
Also, I'm not sure if this is related, but I've just noticed this, and I have no overlays showing...

![image](https://user-images.githubusercontent.com/77614536/219902857-e54db1a5-e632-4cb9-91f0-6a01660b0655.png)

**Comment by Alex-developer** [comment]
@Jonk2 

I have moved all of the modules to /opt/allsky/modules You can just move the folder from /etc to /opt

You are getting the error because the /opt/allsky folder isnt there

**Comment by Jonk2** [comment]
Ok, I've jumped through some hoops (ftp with root) and copied etc/allsky.... to opt/allsky

I've run the install script again, and it's stopped with this error:

```
Installing allsky_pistatus
==========================
INFO: Installing dependencies
INFO: Installing allsky_pistatus module
cp: cannot create regular file '/opt/allsky/modules/allsky_pistatus.py': Permission denied
ERROR: Installation of allsky_pistatus Failed, Could not copy module from /home/pi/allsky-modules/allsky_pistatus/allsky_pistatus.py to /opt/allsky/modules
```
Do I need to run sudo ./install.sh then? I'll try that now.

**Comment by Jonk2** [comment]
The answer is no!

`DO NOT run this as root. Run the installer as the same user as AllSky was installed`

**Comment by Alex-developer** [comment]
Permissions are very important. The permissions on the folder in /opt must be the same as they are in /etc

Change the permissions on the /opt/allsky folder (and all folder/files within) to the user you installed allsky as and www-date

so something like 

chown -R pi:www-data /opt/allsky

**Comment by Jonk2** [comment]
Doesn't seem to work:

```
pi@rorpi:~/allsky-modules $ chown -R pi:www-data /opt/allsky
chown: changing ownership of '/opt/allsky/modules': Operation not permitted
chown: changing ownership of '/opt/allsky': Operation not permitted

```

etc/allsky is pi www-data


![image](https://user-images.githubusercontent.com/77614536/219979002-ef675397-817c-45db-bf6c-7d0b07f829b5.png)

EDIT - let me try again as root

EDIT 2 - yes that worked - sorry, I'm not great with Linux

**Comment by Jonk2** [comment]
Success! Thanks for your help.

![image](https://user-images.githubusercontent.com/77614536/219979138-8ebf0126-f9b1-43b7-aea4-596981570b8c.png)

**Comment by Jonk2** [comment]
Ok I'm confused. Does this automatically now show in the module editor (I have rebooted after installation successful) or do I need to do something else?

**Comment by Alex-developer** [comment]
It will appear in the module editor but only in the periodic flow

**Comment by Jonk2** [comment]
Sorry, not for me it doesn't!

![image](https://user-images.githubusercontent.com/77614536/219980102-ae9e9233-1e15-45d2-abc3-3a4f6f87b2af.png)

**Comment by Alex-developer** [comment]
Doesnt for me either on a fresh install - leave it with me

**Comment by Alex-developer** [comment]
In the module settings make sure 'Show experimental' is enabled

**Comment by Jonk2** [comment]
It is, probably by default as I don't recall changing anything?

![image](https://user-images.githubusercontent.com/77614536/220189922-9adfc4e3-ced4-4fed-9c3f-76ba4cc4d480.png)

**Comment by Jonk2** [comment]
Also there's a typo on that screen "Show exterimental modules". Is that external experimental?!
